System dbus test

89luca89 opened the pull request:
Move from unix socket to dbus communication

This will imply adding a policy to `/usr/share/dbus/system.d` to allow `wheel` users to open a channel in system dbus, in order for a root process (either `t-u-notifier` itself, or `tukit` directly) to send a message on system dbus and trigger the action on the user-facing daemon

Vogtinator commented on dbus-notification.sh:
```diff
@@ -0,0 +1,7 @@
+gdbus call --system   \
```
What is this script for? `call` looks wrong here

89luca89 commented on daemon.go:
```diff
@@ -61,49 +54,25 @@ func notifySend(input string) {
+	c := make(chan *dbus.Signal, 10)
+	conn.Signal(c)
+	for v := range c {
+		if v.Name == Iface+".Notify" {
```
Done thanks! :+1:

Vogtinator commented:
Great, pretty much what I had in mind!

There's just one non-obvious security issue: Currently the client listens for signals from any sender, so other users could also emit this `org.opensuse.tukit.Updated` signal on the system bus. To prevent that, the `client` has to register itself under a well known name and the `daemon` also match the sender. To get such a well known name, create a bus config file like `/usr/share/dbus-1/system.d/org.opensuse.tukit.Updated.conf` (or something with `notifier` inside to not conflict with a possible tukit implementation) that includes

```
  <policy user="root">
    <allow own="org.opensuse.tukit.Updated"/>
  </policy>
```

Vogtinator commented:
Yep

89luca89 commented:
> org.opensuse.tukit.Updated

Cool, I've added it, but right now from my local tests, my normal user can still emit that signal anyway, am I missing something?

Vogtinator commented:
> > org.opensuse.tukit.Updated
> 
> Cool, I've added it, but right now from my local tests, my normal user can still emit that signal anyway, am I missing something?

You'll have to make sure that the `client` registers with that service name (like https://github.com/godbus/dbus/blob/4b691ce883d536a867587e2fcfec1245ebddecd0/_examples/server.go#L48) and that the `daemon` also uses `WithMatchSender("org.opensuse.tukit.Updated")` to only accept it from registered services.

89luca89 commented:
> > > org.opensuse.tukit.Updated
> > 
> > 
> > Cool, I've added it, but right now from my local tests, my normal user can still emit that signal anyway, am I missing something?
> 
> You'll have to make sure that the `client` registers with that service name (like https://github.com/godbus/dbus/blob/4b691ce883d536a867587e2fcfec1245ebddecd0/_examples/server.go#L48) and that the `daemon` also uses `WithMatchSender("org.opensuse.tukit.Updated")` to only accept it from registered services.

Right, done :+1:

Vogtinator commented on daemon.go:
```diff
@@ -61,49 +53,26 @@ func notifySend(input string) {
-	if err != nil {
-		log.Fatal("listen error:", err)
+	if err = conn.AddMatchSignal(
+		dbus.WithMatchInterface(Iface),
```
Nitpick: The usual order is sender, path, iface, member

89luca89 replied:
done thanks :smile:

89luca89 commented:
@Vogtinator 
If all seems good, I'll go ahead and merge

Vogtinator reviewed (approved):
It does! IMO ready to submit and create the probably necessary security review.

Because the `org.opensuse.tukit` prefix is involved I'd also suggest to coordinate with @laenion, maybe on the security review bug.

89luca89 commented:
> It does! IMO ready to submit and create the probably necessary security review.
> 
> Because the `org.opensuse.tukit` prefix is involved I'd also suggest to coordinate with @laenion, maybe on the security review bug.

Cool thanks for the review :+1:
